libs/src/services/creatorNode/CreatorNodeSelection.js

```diff
@@ -185,16 +185,22 @@ class CreatorNodeSelection extends ServiceSelection {
 
     // If not enough secondaries returned from services, select from backups
     remainingSecondaries = this.numberOfNodes - 1 - secondaries.length
-    while (remainingSecondaries-- > 0) {
+    while (remainingSecondaries > 0) {
       const backup = this.selectFromBackups()
-      if (backup && backup !== primary) secondaries.push(backup)
+      if (backup && backup !== primary) {
+        secondaries.push(backup)
+        remainingSecondaries--
+      }
     }
 
     // If not enough secondaries returned from backups, select from unhealthy
```

Comment:
larger point but is selecting from unhealthy the right move?

Reply:
imo, selecting from healthy is better than getting a null secondary. although I think if we get to that point, the client will display some sort of error/read mode only if this is the case